web/src/user_group_popover.ts

```diff
@@ -98,16 +98,32 @@ export function toggle_user_group_info_popover(
                         .get_direct_subgroups_of_group(group)
                         .sort(user_group_components.sort_group_member_name),
                 );
+                const members = sort_group_members(fetch_group_members([...group.members]));
+                const all_individual_members = [...user_groups.get_recursive_group_members(group)];
+                const has_bots =
+                    group.is_system_group &&
+                    all_individual_members.some((member_id) => {
+                        const member = people.get_user_by_id_assert_valid(member_id);
+                        return people.is_active_user_for_popover(member.user_id) && member.is_bot;
+                    });
+                const displayed_subgroups = subgroups.slice(0, 30);
+                const displayed_members =
+                    subgroups.length < 30 ? members.slice(0, 30 - subgroups.length) : [];
                 const args = {
                     group_name: user_groups.get_display_group_name(group.name),
                     group_description: group.description,
-                    members: sort_group_members(fetch_group_members([...group.members])),
+                    members,
                     subgroups,
```

Comment:
Now `members` and `subgroups` are only checked to use whether user has any direct members or direct subgroups, so I think that can either be done using the `displayed_members` and `displayed_subgroups` args or by just passing a new boolean arg.